bothan-binance/src/worker/asset_worker.rs

```diff
@@ -123,7 +179,18 @@ async fn handle_reconnect(
 
             let ids_vec = ids.into_iter().collect::<Vec<String>>();
 
-            if subscribe(&ids_vec, connection).await.is_ok() {
+            if ids_vec.is_empty() {
+                info!("no ids to resubscribe to");
+                return;
+            }
+
+            let packet_id = random();
+
+            if connection
+                .unsubscribe_mini_ticker_stream(packet_id, &ids_vec)
+                .await
+                .is_ok()
+            {
                 info!("resubscribed to all ids");
```

Comment:
I'm confused, shouldn't it be `connection.resubscribe...` or unsub then sub, I only see the unsubscribe function

Reply:
this should be subscribe, fixed